var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -238,6 +264,22 @@ class Dealii(CMakePackage, CudaPackage):
                   msg='To enable {0} it is necessary to build deal.II with '
                       'MPI support enabled.'.format(p))
 
+    # Optional dependencies:
+    conflicts('+adol-c', when='^trilinos+chaco',
+              msg='symbol clash between the ADOL-C library and '
+                  'Trilinos SEACAS Chaco.')
+
+    conflicts('+slepc', when='~petsc',
+              msg='It is not possible to enable slepc interfaces '
+                  'without petsc.')
+
+    @property
+    def generator(self):
+        if '+ninja' in self.spec:
+            return 'Ninja'
+        else:
+            return 'Unix Makefiles'
```

Comment:
Does the choice of generator affect what gets built? I don't think it's wise to allow a different spec/hash for the same binary installation.

Reply:
That's why we didn't add a variant to the base class.

Reply:
Oh yeah, I didn't think about that. Thanks for pointing this out! I'll follow what's done in the other packages and just hardcode
```
generator = 'Ninja'
```